week2/Week2.py

```diff
@@ -0,0 +1,140 @@
+# Task 1 : Checking type of number
+
+def fizz_buzz(num):
+    if num % 3 == 0 and num % 5 == 0:
+        print(num, "-", "Fizz Buzz")
+    elif num == 1 or num == 2 or num == 3 or num == 5:
+        print(num, "-", "Prime")
+    elif num % 5 == 0 and num != 5:
+        print(num, "-", "Buzz")
+    elif num % 3 == 0 and num != 3:
+        print(num, "-", "Fizz")
+    elif num % 2 != 0:
+        for i in range(3, 100):
+            if (num % i) != 0:
+                print(num, "-", "Prime")
+            break
+    else:
+        print(num)
+
+
+fizz_buzz(9)
+
+# Task 2 : try and except error for index
+
+
+try:
+    random = [5, 10, 20]
+    print(random[3])
+except IndexError as e:
+    print(e)
+
+
+# Task 3 : Create a class of Jet inventory with two arguments i.e name and country. Also add the minimum 2 items in
+# the class and print them.
+
+class Jet:
+    def __init__(self, name, country):
+        self.jet_name = name
+        self.country = country
+
+    def details(self):
+        print(self.jet_name, self.country)
+
+
+new_jet = Jet("Jet Airways", "India")
+new_jet.details()
+
+# Task 5 - Write a Python script to check whether a given key already exists in a dictionary.
+
+SongInfo = {"1": "album", "2": "rating", "3": "singer", "4": "writer", "5": "musician",
+            "6": "released", "7": "genre", "8": "duration", "9": "recorded", "10": "producer"}
+for song in SongInfo:
+    print(song, ":", SongInfo[song])
+
+
+def guessKeyValue(key, value):
+    length = len(SongInfo)
+    for element in SongInfo:
+        length -= 1
+        if element == key:
+            if SongInfo[element] == value:
+                print("True - ", element, ",", SongInfo[element])
+                break
+            else:
+                print("False - Incorrect value of ", element)
+                break
+        else:
+            if length == 0:
+                print("False, No such key found!")
+            else:
+                continue
```

Comment:
Ok, there's a lot of stuff not right here. First, I will give you some general tips and then some hints towards a better solution.

If you use `continue`or `break` in your loops, you need a very good reason for that. Otherwise, these are considered bad practice. `continue`and `break` are bad because if someone else reads your code, those usually show that you are handling some corner case. corner cases are not general, in programming we like if things are general. if you happen to use `break`and `continue` there is usually a better solution out there.

Next, both of the `break` expressions in your code are not necessary, because in both cases after the `break` we are at the end of the scope of the loop anyway.

Next, the question only asks for checking the key, not the value. By also checking for the value, you overcomplicate things for you.

In Python one can loop elegantly over a dictionary like this:
```Python
for key,val in SongInfo.items():
   print(key,val)

for key in SongInfo.keys():
   print(key)

for val in SongInfo.values():
   print(val)
```

Apart from that, a very easy way to check if a key is in a dictionary:
```Python
a = {1:"test", 2:"test2"}
print(1 in a)
print(3 in a)
```

Try to use all the above to improve your code. Don't check for the value in the dictionary

Reply:
Thanks for clearing this doubt. I was bit confused in using continue and break but good that I tried here and got valuable information. I have implemented all as per your feedback.